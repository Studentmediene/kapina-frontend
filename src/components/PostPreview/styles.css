.image {
  width: 100%;
}

.title {
  text-align: center;
  margin-top: 0.7rem;
  margin-bottom: 0.5rem;
}

.titleLink {
  text-decoration: none;
  color: inherit;
}

.titleLink:hover {
  text-decoration: underline;
}

.createdBy {
  text-align: center;
  color: gray;
  font-size: smaller;
}

.lead {
  margin-top: 0;
}

.category {
  position: absolute;
  margin-left: -0.5em;
  padding: 0.1em 0.5em;
  border-radius: 0.3em;
}
