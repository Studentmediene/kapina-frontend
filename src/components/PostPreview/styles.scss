@import 'variables.scss';

.image {
  width: 100%;
}

.title {
  font-size: $frontpage-title-size;
  margin-bottom: 0.5rem;
  margin-top: 0.7rem;
  text-align: center;
}

.titleLink {
  color: inherit;
  text-decoration: none;
}

.titleLink:hover {
  text-decoration: underline;
}

.createdBy {
  color: $color-text-dark;
  font-size: smaller;
  text-align: center;
}
