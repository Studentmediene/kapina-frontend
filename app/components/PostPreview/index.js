/**
*
* PostPreview
*
*/

import React from 'react';
import { Link } from 'react-router';

import styles from './styles.css';

function PostPreview(props) {
  return (
    <div className={styles.postPreview}>
      <Link className={styles.imageLink} to={`/post/${props.slug}`}>
        <img className={styles.image} src={props.coverPhoto} alt={props.title} />
      </Link>
      <Link className={styles.titleLink} to={`/post/${props.slug}`}>
        <h2 className={styles.title}>
          {props.title}
        </h2>
      </Link>
      <p className={styles.lead}>{props.lead}</p>
    </div>
  );
}

PostPreview.propTypes = {
  title: React.PropTypes.string,
  slug: React.PropTypes.string,
  lead: React.PropTypes.string,
  coverPhoto: React.PropTypes.string,
};

export default PostPreview;
