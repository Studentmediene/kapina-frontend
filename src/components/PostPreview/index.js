import React from 'react';
import PropTypes from 'prop-types';
import LazyLoad from 'react-lazyload';
import { Link } from 'react-router-dom';

import styles from './styles.css';

const PostPreview = props => {
  let categories;
  if (props.categories && props.categories.length > 0) {
    categories = props.categories.map((category, index) => (
      <div
        key={category.name}
        className={styles.category}
        style={{
          marginTop: `-${(index + 1) * 2}em`,
          backgroundColor: `${category.backgroundColor}`,
          color: `${category.textColor}`,
        }}
      >
        {category.name}
      </div>
    ));
  }

  return (
    <div className={styles.postPreview}>
      <Link className={styles.imageLink} to={`/post/${props.slug}`}>
        <LazyLoad height={350} offset={100} once>
          <img
            className={styles.image}
            src={props.coverPhotoUrl}
            alt={props.title}
          />
        </LazyLoad>
        {categories}
      </Link>
      <Link className={styles.titleLink} to={`/post/${props.slug}`}>
        <h2 className={styles.title}>{props.title}</h2>
      </Link>
      <p className={styles.lead}>{props.lead}</p>
    </div>
  );
};

PostPreview.propTypes = {
  title: PropTypes.string.isRequired,
  slug: PropTypes.string.isRequired,
  lead: PropTypes.string.isRequired,
  coverPhotoUrl: PropTypes.string,
  categories: PropTypes.array,
};

export default PostPreview;
