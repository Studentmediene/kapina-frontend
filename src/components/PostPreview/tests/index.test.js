import React from 'react';
import { shallow } from 'enzyme';

import PostPreview from '../';

describe('<PostPreview />', () => {
  it('renders correctly with categories', () => {
    const mockProps = {
      croppedImages: {
        large: 'large_image',
        medium: 'medium_image',
        small: 'small_image',
      },
      lead: 'Lead',
      publishAt: '2017-11-01 18:11:44+00:00',
      slug: 'slug',
      title: 'title',
      categories: [
        {
          backgroundColor: '#FFFFFF',
          name: 'categoryName1',
          textColor: '#000000',
        },
        {
          backgroundColor: '#FFFFFF',
          name: 'categoryName1',
          textColor: '#000000',
        },
      ],
    };
    const tree = shallow(<PostPreview {...mockProps} />);
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly without categories', () => {
    const mockProps = {
      croppedImages: {
        large: 'large_image',
        medium: 'medium_image',
        small: 'small_image',
      },
      lead: 'Lead',
      publishAt: '2017-11-01 18:11:44+00:00',
      slug: 'slug',
      title: 'title',
    };
    const tree = shallow(<PostPreview {...mockProps} />);
    expect(tree).toMatchSnapshot();
  });
});
