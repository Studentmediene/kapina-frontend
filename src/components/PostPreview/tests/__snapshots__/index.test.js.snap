// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostPreview /> renders correctly with categories 1`] = `
<ImageLink
  as="/post/slug"
  className="link"
  href="/post/[slug]"
  imageDescription=""
  imageProps={Object {}}
  images={
    Object {
      "large": "large_image",
      "medium": "medium_image",
      "small": "small_image",
    }
  }
>
  <div
    className="categoryContainer"
  >
    <CategoryTag
      backgroundColor="#FFFFFF"
      index={0}
      key="0"
      name="categoryName1"
      position="bottom"
      textColor="#000000"
    />
    <CategoryTag
      backgroundColor="#FFFFFF"
      index={1}
      key="1"
      name="categoryName1"
      position="bottom"
      textColor="#000000"
    />
  </div>
  <h2
    className="title"
  >
    title
  </h2>
</ImageLink>
`;

exports[`<PostPreview /> renders correctly without categories 1`] = `
<ImageLink
  as="/post/slug"
  className="link"
  href="/post/[slug]"
  imageDescription=""
  imageProps={Object {}}
  images={
    Object {
      "large": "large_image",
      "medium": "medium_image",
      "small": "small_image",
    }
  }
>
  <div
    className="categoryContainer"
  />
  <h2
    className="title"
  >
    title
  </h2>
</ImageLink>
`;
